File: reference/go.rb
#! /usr/bin/env ruby

a = 5

b = a

puts "a is #{a} and b is #{b}"

puts "a object id is: #{a.object_id}  class is #{a.class}"
puts "b object id is: #{b.object_id}  class is #{b.class}"

b = a.next

puts "a is #{a} and b is #{b}"

puts "a object id is: #{a.object_id}  class is #{a.class}"
puts "b object id is: #{b.object_id}  class is #{b.class}"


puts "\ndowncase"

sa = "ABC"

sb = sa

puts "sa is #{sa} and sb is #{sb}"

puts "sa object id is: #{sa.object_id}  class is #{sa.class}"
puts "sb object id is: #{sb.object_id}  class is #{sb.class}"

sb = sa.downcase

puts "a is #{sa} and b is #{sb}"

puts "sa object id is: #{sa.object_id}  class is #{sa.class}"
puts "sb object id is: #{sb.object_id}  class is #{sb.class}"

puts "\ndowncase!"

sa = "ABC"

sb = sa

puts "sa is #{sa} and sb is #{sb}"

puts "sa object id is: #{sa.object_id}  class is #{sa.class}"
puts "sb object id is: #{sb.object_id}  class is #{sb.class}"

sb = sa.downcase!

puts "sa is #{sa} and sb is #{sb}"

puts "sa object id is: #{sa.object_id}  class is #{sa.class}"
puts "sb object id is: #{sb.object_id}  class is #{sb.class}"

File: YAML/05_dist_prototype/go.py
#! /usr/bin/env ruby

require 'yaml'
require 'pp'

puts "hello"



class MoveDirectories

  def initialize(config)
    @directories = Hash.new()
    config.each do | dir,conditions |
      if conditions.nil?
        @directories[dir] = ['.*']
      else
        @directories[dir] = conditions
      end
    end
  end

  def resolve(product_path)

    filename = File::basename(product_path)

    paths = Array.new()
    
    @directories.each do |dir, conditions |
      conditions.each do | condition |
        if Regexp.new(condition).match filename
          paths << dir
          break
        end
      end
    end

    return paths
  end

end

configure = YAML::load_file('move_dirs.yaml')

move_config = configure["move_directories"]

move_paths = MoveDirectories.new(move_config)

products = ["ops.skewT.20120424110300.profile.jpg",
            "reseach.gv.20120424110300.path.kml",
            "report.ship.20120424110300.plan.pdf",
            "ops.radar.20120424110300.dbz.jpg",
            "ops.radar.20120424110300.dbz.jpg.gz"]

products.each do | product |
  puts product
  paths = move_paths.resolve(product)
  paths.each do | path |
    puts "  " + path
  end
  puts ''
end


puts "bye"
